dokka-subprojects/analysis-kotlin-symbols/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/symbols/translators/AnnotationTranslator.kt

```diff
@@ -57,10 +58,17 @@ internal class AnnotationTranslator {
     }
 
     private fun KaSession.mustBeDocumented(annotation: KaAnnotation): Boolean {
-        if (annotation.isNoExistedInSource()) return false
+        /**
+         * Do not document the synthetic [parameterNameAnnotation] annotation since Dokka K1 ignores it too.
+         * The annotation can be added by the compiler during "desugaring" functional types.
+         * e.g., `(x: Int) -> Unit` becomes `Function1<@ParameterName("x") Int, Unit>`
+         * @see ParameterName
+         * @see getPresentableName
+        */
+        if(annotation.classId == parameterNameAnnotation && annotation.isNoExistedInKotlinSource()) return false
```

Comment:
I have not discovered other synthetic annotations added by the compiler.  
The PR aims to keep the K1 behaviour that might be reconsidered later.

JIC I have opened https://youtrack.jetbrains.com/issue/KT-70325/Analysis-API-KaAnnotation-returns-only-Kotlin-PSI but it does not affect us now.